rclcpp/src/rclcpp/duration.cpp

```diff
@@ -182,6 +182,34 @@ Duration::operator-(const rclcpp::Duration & rhs) const
     rcl_duration_.nanoseconds - rhs.rcl_duration_.nanoseconds);
 }
 
+void
+bounds_check_duration_scale(int64_t dns, double scale, uint64_t max)
+{
+  auto abs_dns = static_cast<uint64_t>(std::abs(dns));
+  auto abs_scale = std::abs(scale);
```

Comment:
This is apparently ambiguous on OSX: https://ci.ros2.org/job/ci_osx/3306/consoleFull

```
03:04:22 [ 38%] Building CXX object CMakeFiles/rclcpp.dir/src/rclcpp/contexts/default_context.cpp.o
03:04:23 [ 39%] Building CXX object CMakeFiles/rclcpp.dir/src/rclcpp/duration.cpp.o
03:04:23 /Users/osrf/jenkins/workspace/ci_osx/ws/src/ros2/rclcpp/rclcpp/src/rclcpp/duration.cpp:189:20: error: call to 'abs' is ambiguous
03:04:23   auto abs_scale = std::abs(scale);
03:04:23                    ^~~~~~~~
03:04:23 /usr/include/stdlib.h:137:6: note: candidate function
03:04:23 int      abs(int) __pure2;
03:04:23          ^
03:04:23 /Library/Developer/CommandLineTools/usr/include/c++/v1/stdlib.h:115:44: note: candidate function
03:04:23 inline _LIBCPP_INLINE_VISIBILITY long      abs(     long __x) _NOEXCEPT {return  labs(__x);}
03:04:23                                            ^
03:04:23 /Library/Developer/CommandLineTools/usr/include/c++/v1/stdlib.h:117:44: note: candidate function
03:04:23 inline _LIBCPP_INLINE_VISIBILITY long long abs(long long __x) _NOEXCEPT {return llabs(__x);}
03:04:23                                            ^
03:04:23 /Users/osrf/jenkins/workspace/ci_osx/ws/src/ros2/rclcpp/rclcpp/src/rclcpp/duration.cpp:203:13: error: no member named 'isnormal' in namespace 'std'
03:04:23   if (!std::isnormal(scale)) {
03:04:23        ~~~~~^
03:04:23 2 errors generated.
03:04:23 make[2]: *** [CMakeFiles/rclcpp.dir/build.make:212: CMakeFiles/rclcpp.dir/src/rclcpp/duration.cpp.o] Error 1
03:04:23 make[1]: *** [CMakeFiles/Makefile2:855: CMakeFiles/rclcpp.dir/all] Error 2
03:04:23 make: *** [Makefile:141: all] Error 2
03:04:23 
```

Reply:
You want `std::fabs`: http://en.cppreference.com/w/cpp/numeric/math/fabs

Reply:
Though, it should probably work anyways...

Reply:
You might just need to `#include <cmath>`. The standard seems to have been inconsistent on that in the past, which would allow for two different implementations to give you different results.

Reply:
Ok, I'll explicitly add #include <cmath>, thank you